### PR TITLE
Refactor lib to TS

### PR DESCRIPTION
**Pull request by khaosdoctor**
Refactoring lib to TS, no breaking changes, the function signatures are still the same.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
   "version": "1.5.2",
   "description": "Sugar code to read environment variables with default values.",
-  "main": "index.js",
+  "main": "dist/index.js",
   "scripts": {
     "coverage": "nyc --branches 100 --functions 100 --lines 100 --statements 100 npm test",
     "lint": "standard --global describe --global it",
-    "test": "mocha test/*.test.js --check-leaks --full-trace --use_strict --recursive"
+    "test": "mocha test/*.test.js --check-leaks --full-trace --use_strict --recursive",
+    "build": "tsc"
```
Comment (roziscoding) [inline package.json:11]: There should also be a `prepare` script that runs `tsc` so we don't forget to transpile before publishing

**Review by roziscoding — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (khaosdoctor) [reply package.json:11]: Solved!

**Review by roziscoding — APPROVED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+type Nullable<T> = T | null
+
+interface IConversionFunction<T> {
+  (value: string): Nullable<T>
+}
+
+function _getAs<T> (fn: IConversionFunction<T>) {
+  return (names: string | string[], fallback: Nullable<string>) => {
+    const value = get(names, fallback)
+    if (!value) return value
```
Comment (rwillians) [inline src/index.ts]: This is testing agains a `falsy` value but the current behaviour is testing only against `null` values.
This causes the following breaking change:

Current implementation:
```js
// process.env.FOOBAR = ''
env.get.int('FOOBAR', 0) // Number(0)
env.get.int('FOOBAR', '0') // Number(0)
env.get.int('FOOBAR', false) // NaN
env.get.int('FOOBAR', true) // NaN
```

Proposed implementation:
```js
// process.env.FOOBAR = ''
env.get.int('FOOBAR', 0) // Number(0)
env.get.int('FOOBAR', '0') // Number(0)
env.get.int('FOOBAR', false) // Boolean(false)
env.get.int('FOOBAR', true) // NaN
```

There's room for adding an unitary test for this you if may.

**Review by rwillians — CHANGES_REQUESTED** [review]

**Comment by rwillians** [comment]
@khaosdoctor @rjmunhoz except for this breaking change that I've pointed out, LGTM. Thanks for contributing.

File: src/index.ts (hunk 2 above)
Comment (khaosdoctor) [reply src/index.ts]: Hum! Ok I Got it! Will correct here and add a test to it

Comment (rwillians) [reply src/index.ts]: Great

Comment (khaosdoctor) [reply src/index.ts]: Fixed! in 4341708a81d3864fbc9d37d2a43a13318957cc2b

**Comment by khaosdoctor** [comment]
I have updated the lib and added some tests and corrected this behavior you pointed out :D

**Comment by rwillians** [comment]
Great @khaosdoctor, thanks!

**Comment by rwillians** [comment]
@khaosdoctor I'll ping you when it's published

**Pull request merged**